File: sdk/samples/WPFSamples/FixedLinearTopologySample/visualbasic/wizardlauncher.vb
Public Class WizardLauncher
    Inherits PageFunction(Of WizardContext)

    Public Sub New()
        Me.WizardData = New WizardData
    End Sub

    Protected Overrides Sub Start()
        MyBase.Start()
        MyBase.KeepAlive = True
        Dim firstPage As New WizardPage1(Me.WizardData)
        AddHandler firstPage.Return, New ReturnEventHandler(Of WizardResult)(AddressOf Me.WizardPage_Return)
        MyBase.NavigationService.Navigate(firstPage)
    End Sub

    Public Sub WizardPage_Return(ByVal sender As Object, ByVal e As ReturnEventArgs(Of WizardResult))
        Me.OnReturn(New ReturnEventArgs(Of WizardContext)(New WizardContext(e.Result, Me.WizardData)))
    End Sub

    Private WizardData As WizardData

End Class

File: sdk/samples/WPFSamples/WizardSample/csharp/wizardlauncher.cs
namespace WizardSample
{
    using System;
    using System.Windows;
    using System.Windows.Controls;
    using System.Windows.Navigation;

    public class WizardLauncher : PageFunction<WizardResult>
    {
        WizardData wizardData = new WizardData();
        public event WizardReturnEventHandler WizardReturn;

        protected override void Start()
        {
            base.Start();

            // So we remember the WizardCompleted event registration
            this.KeepAlive = true;
            
            // Launch the wizard
            WizardPage1 wizardPage1 = new WizardPage1(this.wizardData);
            wizardPage1.Return += new ReturnEventHandler<WizardResult>(wizardPage_Return);
            this.NavigationService.Navigate(wizardPage1);
        }

        public void wizardPage_Return(object sender, ReturnEventArgs<WizardResult> e)
        {
            // Notify client that wizard has completed
            // NOTE: We need this custom event because the Return event cannot be
            // registered by window code - if WizardDialogBox registers an event handler with
            // the WizardLauncher's Return event, the event is not raised.
            if (this.WizardReturn != null)
            {
                this.WizardReturn(this, new WizardReturnEventArgs(e.Result, this.wizardData));
            }
            OnReturn(null);
        }
    }
}
